src/App.tsx

```diff
@@ -1,14 +1,57 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { ChangeEvent, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+// import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setQuery] = useState('');
+  const [filteredTodos, setFilteredTodos] = useState(todos);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onInputQuery = (event: ChangeEvent<HTMLInputElement>) => setQuery(event.target.value);
+
+  const onClearInput = () => setQuery('');
+
+  const visibleTodos = filteredTodos.filter(todo => {
+    const lowerTitle = todo.title.toLowerCase();
+    const lowerQuery = searchQuery.toLowerCase().trim();
+
+    return lowerTitle.includes(lowerQuery);
+  });
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(data => {
+        setTodos(data);
+        setFilteredTodos(data);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const onSelectTodos = (event: ChangeEvent<HTMLSelectElement>) => {
```

Comment:
[naming](https://medium.com/swlh/how-to-better-name-your-functions-and-variables-e962a4ef335b)